hr_holidays_public/__manifest__.py

```diff
@@ -14,7 +14,7 @@
     "Odoo Community Association (OCA),",
     "summary": "Manage Public Holidays",
     "website": "https://github.com/OCA/hr-holidays",
-    "depends": ["hr_holidays"],
+    "depends": ["hr_holidays", "base_location"],
```

Comment:
Maybe not everybody is happy to pull this new dependency...

Reply:
Totally agree, but then I think it would be necessary to add a new module with both dependencies (`hr_holidays_public` + `base_location`).

The model res.city is added with `base_address_city`
`city_id` field in `res.company` is added with `base_location`

Reply:
AFAIK, you don't need res.company->city_id. You need res.partner->city_id. So the new dependency is `base_address_city`, not `base_location`. Adding such dependency on a stable module? Well, it can be debatable, but it's not a big deal IMO, as it's a core Odoo module that is also very transparent once installed.

Let's see what other contributors like @ivantodorovich or @petrus-v think about it.

Reply:
I would prefer to have a glue module for this, rather than adding the new dependency, even if it's `base_address_city`

Reply:
Now the dependency added is `base_address_city`, so do you think this functionality should go in a new module?

Reply:
I'm not using `base_address_city` that's much and I'm learning that there are cities that gives more days off (I'm wondering which ones ? I'm so curious to know, where should I work to get more days off, we could open a postal box for remote employes there :p).

So technically spoken I would prefer a new module that depends on both `hr_holidays_public` + `base_address_city` to avoid change on res partner view and maintains code that I do not really needs.

In an other way as long `base_address_city` appears to be transparent for end user and manage case where city_id is not used it's kind of ok with probably a roadmap to suggest to refactor the module ! but I would say it's pity to not do it first !

my  2cts

Reply:
For me i think it's not a big deal to add this dependency and i think the improvement is a good one. So in my opinion i'm okay with this change, and also will like to see it ported to 14.0 version since it seems very useful for some use cases.

Reply:
>  that's much and I'm learning that there are cities that gives more days off

I don't know of other countries, but in Spain there are days allocated by the nation, days allocated by the regions/states and days allocated by the municipalities; so every spanish has the same holidays but they can be on different days depending on where they live.

Reply:
> > that's much and I'm learning that there are cities that gives more days off
> 
> I don't know of other countries, but in Spain there are days allocated by the nation, days allocated by the regions/states and days allocated by the municipalities; so every spanish has the same holidays but they can be on different days depending on where they live.

ok good to know ! where employee lives or according the office place ? (As far I've understood source code it's get from the employee office place !)

Reply:
@petrus-v Not always. The business can have people working in a lot of different cities in the country, some remote, some on-site. So sometimes is according to where the employee lives. Also, an employee who works in a office, probably the office (the one he need to attend to) is in the same city so office city and employee city usually to match.